src/xdocs-examples/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/naming/patternvariablename/Example1.java

```diff
@@ -0,0 +1,23 @@
+/*xml
+<module name="Checker">
+  <module name="TreeWalker">
+    <module name="PatternVariableName"/>
+  </module>
+</module>
+*/
+
+//non-compiled with javac: Compilable with Java17
+package com.puppycrawl.tools.checkstyle.checks.naming.patternvariablename;
+
+// xdoc section -- start
+class Example1 {
+  Example1(Object o1){
+    if (o1 instanceof String STRING) { // violation
+    }
```

Comment:
please make code like `if (o1 instanceof String STRING) {} // violation`
atention to `{}`  to let user not bother wtih block at all.